packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/common.ts

```diff
@@ -58,7 +76,104 @@ export interface CommonDestinationProps {
    * @default - if `logging` is set to `true`, a log group will be created for you.
    */
   readonly logGroup?: logs.ILogGroup;
+}
 
+/**
+ * Common properties for defining a backup, intermediary, or final S3 destination for a Kinesis Data Firehose delivery stream.
+ */
+export interface CommonS3Props {
+  /**
+   * The length of time that Firehose buffers incoming data before delivering
+   * it to the S3 bucket.
+   *
+   * If bufferingInterval is specified, bufferingSize must also be specified.
+   * Minimum: Duration.seconds(60)
+   * Maximum: Duration.seconds(900)
+   *
+   * @default Duration.seconds(300)
+   */
+  readonly bufferingInterval?: cdk.Duration;
+
+  /**
+   * The size of the buffer that Kinesis Data Firehose uses for incoming data before
+   * delivering it to the S3 bucket.
+   *
+   * If bufferingSize is specified, bufferingInterval must also be specified.
+   * Minimum: Size.mebibytes(1)
+   * Maximum: Size.mebibytes(128)
+   *
+   * @default Size.mebibytes(5)
+   */
+  readonly bufferingSize?: cdk.Size;
+
+  /**
+   * The type of compression that Kinesis Data Firehose uses to compress the data
+   * that it delivers to the Amazon S3 bucket.
+   *
+   * The compression formats SNAPPY or ZIP cannot be specified for Amazon Redshift
+   * destinations because they are not supported by the Amazon Redshift COPY operation
+   * that reads from the S3 bucket.
+   *
+   * @default - UNCOMPRESSED
+   */
+  readonly compression?: Compression;
+
+  /**
+   * The AWS KMS key used to encrypt the data that it delivers to your Amazon S3 bucket.
+   *
+   * @default - Data is not encrypted.
+   */
+  readonly encryptionKey?: kms.IKey;
+
+  /**
+   * A prefix that Kinesis Data Firehose evaluates and adds to failed records before writing them to S3.
+   *
+   * This prefix appears immediately following the bucket name.
+   * @see https://docs.aws.amazon.com/firehose/latest/dev/s3-prefixes.html
+   *
+   * @default "YYYY/MM/DD/HH"
+   */
+  readonly errorOutputPrefix?: string;
+
+  /**
+   * A prefix that Kinesis Data Firehose evaluates and adds to records before writing them to S3.
+   *
+   * This prefix appears immediately following the bucket name.
+   * @see https://docs.aws.amazon.com/firehose/latest/dev/s3-prefixes.html
+   *
+   * @default "YYYY/MM/DD/HH"
+   */
+  readonly dataOutputPrefix?: string;
+}
+
+/**
+ * Properties for defining an S3 backup destination.
+ *
+ * S3 backup is available for all destinations, regardless of whether the final destination is S3 or not.
+ */
+export interface S3BackupDestinationProps extends DestinationLoggingProps, CommonS3Props {
```

Comment:
So, since this interface is used in `CommonDestinationProps`, I wonder whether `DestinationS3BackupProps` is not a better name. "S3BackupDestinationProps"  to me sounds like the "S3 backup" part is the destination itself, similarly like I could name something an "S3Destination" (I understand you don't use this convention, but I think the point stands).

Reply:
Aha, you raise a great point. My mental model was influenced by the fact that the S3Backup configuration in the Cloudformation API **is-a** S3Destination. So I was thinking it was an "S3 ... BackupDestination" not an "S3Backup ... Destination". Since this is an attribute of the Destination, and not a destination in and of itself. I agree with the names you propose here.